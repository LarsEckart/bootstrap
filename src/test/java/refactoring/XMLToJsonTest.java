package refactoring;

import java.net.URL;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.BeyondCompareMacReporter;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

@UseReporter(BeyondCompareMacReporter.class)
class XMLToJsonTest {

  @Test
  void approval() throws Exception {
    verifyXmlToJson("fk:AMM24_fk:AMM24-FM");
  }

  @Test
  void approval_root() throws Exception {
    verifyXmlToJson("/");
  }

  @Test
  void approval_trnum() throws Exception {
    XMLToJson xmlToJson = new XMLToJson();
    URL url = getClass().getClassLoader().getResource("xmlToJson.xml");

    String json = xmlToJson.getJson(url, "ft:loetr");

    String replace =
        json.replace(",", ",\n")
            .replace("}", "\n}\n")
            .replace("{", "\n{\n");

    Approvals.verify(replace);
  }

  private void verifyXmlToJson(String xPathString) throws Exception {
    XMLToJson xmlToJson = new XMLToJson();
    URL url = getClass().getClassLoader().getResource("xmlToJson.xml");

    String json = xmlToJson.getJson(url, xPathString);

    Approvals.verifyJson(json);
  }
}
