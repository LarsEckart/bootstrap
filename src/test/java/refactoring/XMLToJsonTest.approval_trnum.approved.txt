[
{
'data':'[TR12-118] TASK AMM12-31-00-660-806 - Inspection and Removal of De-Hydrated Anti-Icing Fluid inside the Flight Control Surfaces',

{
'data':'[TR12-118] TASK AMM12-31-00-660-806 - Inspection and Removal of De-Hydrated Anti-Icing Fluid inside the Flight Control Surfaces',
'attr':
{
'id':'ft:loetr_dtrn:TR12-118',
'file':'TR12-118.pdf'
}
,
'state':'closed'
}
]